modules/accredited_representative_portal/app/controllers/concerns/accredited_representative_portal/v0/representative_form_upload_concern.rb

```diff
@@ -102,6 +102,20 @@ def last_name
       def birth_date
         claimant_birth_date || veteran_birth_date
       end
+
+      def validate_power_of_attorney
+        authorize(get_icn, policy_class: RepresentativeFormUploadPolicy)
+      end
+
+      def get_icn
+        mpi = MPI::Service.new.find_profile_by_attributes(ssn:, first_name:, last_name:, birth_date:)
+
+        if mpi.profile&.icn
+          mpi.profile.icn
+        else
+          raise Common::Exceptions::RecordNotFound, 'Could not lookup claimant with given information.'
+        end
+      end
```

Comment:
Maybe this could go back into the controller. The rest of this file actually wants to be a method in a service for doing these form things, so we should halt making use of a controller concern here early.